### DISCUSSION TITLE
Feature request: Retry querying the data source if getting DatasourceError or DatasourceNoData

### DISCUSSION THREAD

**Discussion by erkexzcx** [grafana/grafana#46871] [Alerting]
I am heavily using Grafana's unified alerting and one the most annoying things about it that it immediatelly fires the alert if data source has not been available for a minute, returned non-200 http code or returned empty data.

Can we have configuration for this? Or even better - respect `for` value in Grafana and wait given amount of time, just like it's done with actual data for alerting. As of now I am getting alerts at any time of the day and the amount of false alerts during nights is just making Grafana annoying to use.

![image](https://user-images.githubusercontent.com/5955795/159662436-e039f942-ca9a-4621-8cbc-86dbd1e4202a.png)

**Comment by grigassukys** [comment]
Any updates on this issue and the possibility to have configuration on this?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
